src/circuit/ecc.rs

```diff
@@ -1,42 +1,66 @@
-use crate::rns::Integer;
+use crate::rns::{Integer, Rns};
 
 use super::{integer::IntegerConfig, AssignedInteger};
-use halo2::arithmetic::CurveAffine;
+use crate::circuit::integer::IntegerChip;
+use halo2::arithmetic::{CurveAffine, FieldExt};
 use halo2::circuit::Region;
 use halo2::plonk::Error;
+use num_bigint::BigUint as big_uint;
 
 pub struct Point<C: CurveAffine> {
     x: Integer<C::ScalarExt>,
     y: Integer<C::ScalarExt>,
 }
 
+/// E is emulated curve, C is the native curve
 impl<C: CurveAffine> Point<C> {
-    fn new(x: Integer<C::ScalarExt>, y: Integer<C::ScalarExt>) -> Self {
+    pub fn new_from_point<E: CurveAffine>(p: E, num_of_limbs: usize, bit_len: usize) -> Self {
```

Comment:
It seems like this function can go under `rns` since rns keeps number of limbs and bit lenght. And we would only need this function in pre prover time similar with how we do it in integer tests